src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -388,8 +370,13 @@ public void insertRecordToBuffer(SinkRecord kafkaSinkRecord) {
       this.latestConsumerOffset.set(kafkaSinkRecord.kafkaOffset());
     }
 
-    // Ignore adding to the buffer until we see the expected offset value
-    if (shouldIgnoreAddingRecordToBuffer(kafkaSinkRecord, currentProcessedOffset)) {
+    // Reset the value if it's a new batch
+    if (isFirstRowInBatch) {
+      needToSkipCurrentBatch = false;
+    }
+
+    // Simply skip inserting into the buffer if the row should be ignored after channel reset
+    if (needToSkipCurrentBatch) {
```

Comment:
TODO: add a log here when we're skipping rows

Reply:
are you planning to do it here?